x-pack/plugins/secops/server/lib/framework/types.ts

```diff
@@ -54,6 +66,35 @@ export interface DatabaseSearchResponse<Hit = {}, Aggregations = undefined>
   };
 }
 
-export interface InfraDatabaseMultiResponse<Hit, Aggregation> extends DatabaseResponse {
+export interface DatabaseMultiResponse<Hit, Aggregation> extends DatabaseResponse {
   responses: Array<DatabaseSearchResponse<Hit, Aggregation>>;
 }
+
+interface FrameworkIndexFieldDescriptor {
+  name: string;
+  type: string;
+  searchable: boolean;
+  aggregatable: boolean;
+  readFromDocValues: boolean;
+}
+
+export interface FrameworkIndexPatternsService {
+  getFieldsForWildcard(options: {
+    pattern: string | string[];
+  }): Promise<FrameworkIndexFieldDescriptor[]>;
+}
+
+export interface FrameworkIndexPatternsService {
+  getFieldsForWildcard(options: {
+    pattern: string | string[];
+  }): Promise<FrameworkIndexFieldDescriptor[]>;
+}
+
+export interface DatabaseGetIndicesResponse {
+  [indexName: string]: {
+    aliases: {
+      // tslint:disable-next-line:no-any
+      [aliasName: string]: any;
```

Comment:
I take it the typing of this is large and/or narly but it's not a Hapi Response in which we have types for?